src/main/java/mainpack/Agreement.java

```diff
@@ -17,10 +19,17 @@ public void setNumber(Integer number) {
         this.number = number;
     }
 
-    public void setIncome(Integer income) {
-        this.income = income;
+    public void setIncome() {
+        if(this.subject.getType().equals("House")){
+            this.income = subject.nominalPrice * 0.02 + 250;
+        } else {
+            this.income = subject.nominalPrice * 0.01 + 100;
+        }
```

Comment:
For the future features you should consider to check all the types you have, there should be 3 cases as for now:
if House -> ...
else if Apartment -> ...
else -> throw exception, because we don't have selected type